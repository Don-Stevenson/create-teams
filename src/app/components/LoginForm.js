'use client'

// components/LoginForm.js
import { useState } from 'react'
import { useRouter } from 'next/navigation'
import LoonsBadge from '../assets/img/TWSC.webp'
import Image from 'next/image'
import { login } from '../../../utils/FEapi'

export default function LoginForm() {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const router = useRouter()
  const [error, setError] = useState(false)
  const [isLoading, setIsLoading] = useState(false)

  const handleSubmit = async e => {
    e.preventDefault()
    setIsLoading(true)
    try {
      const data = await login(username, password)
      if (data.success) {
        setError(false)
        router.push('/create-teams')
        router.refresh() // Force a refresh of the page to update authentication state
      } else {
        setError(true)
      }
    } catch (error) {
      setError(true)
      console.error('Login error:', error)
    } finally {
      setIsLoading(false)
    }
  }

  return (
    <form
      onSubmit={handleSubmit}
      data-testid="login-form"
      className="min-h-screen bg-background"
    >
      <div className="flex flex-col items-center justify-center mt-20 gap-2">
        <Image
          width={100}
          height={125}
          src={LoonsBadge}
          alt="Toronto Walking Soccer Loons Club Logo"
          priority
        />
        <div className="flex bg-loonsDarkBrown z-0 w-[248px] justify-center h-[168px] items-center rounded mb-4">
          <div className="flex items-center justify-center text-3xl font-bold border-[8px] border-loonsRed rounded bg-loonsBrown w-60 text-loonsBeige text-center h-40 z-10">
            Loons Team Balancer
          </div>
        </div>
        <div className="flex flex-col justify-center xs:flex-row gap-3 mb-3">
          <input
            type="text"
            value={username}
            onChange={e => setUsername(e.target.value)}
            onFocus={() => setError(false)}
            placeholder="Username"
            required
            disabled={isLoading}
            className="border border-gray-300 rounded w-40 h-8 text-center focus:outline-none focus:ring-2 focus:ring-loonsRed disabled:opacity-50"
          />
          <input
            type="password"
            value={password}
            onChange={e => setPassword(e.target.value)}
            onFocus={() => setError(false)}
            placeholder="Password"
            required
            disabled={isLoading}
            className="border border-gray-300 rounded w-40 h-8 text-center focus:outline-none focus:ring-2 focus:ring-loonsRed disabled:opacity-50"
          />
        </div>
        <div className="flex justify-center items-center">
          <button
            type="submit"
            disabled={isLoading}
            className="border border-gray-400 rounded w-36 h-8 bg-gray-200 hover:bg-gray-300 transition-colors disabled:opacity-50"
          >
            {isLoading ? 'Logging in...' : 'Login'}
          </button>
        </div>
      </div>
      <div className="flex justify-center text-center items-center text-loonsRed h-10">
        {error ? "There's been an error. Please try again" : ''}
      </div>
    </form>
  )
}
