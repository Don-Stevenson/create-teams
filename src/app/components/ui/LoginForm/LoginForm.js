'use client'

// components/LoginForm.js
import { useState } from 'react'
import { useRouter } from 'next/navigation'
import LoonsBadge from '../../../assets/img/TWSC.webp'
import Image from 'next/image'
import { useLogin } from '../../../hooks/useApi'

export default function LoginForm() {
  const [username, setUsername] = useState('')
  const [password, setPassword] = useState('')
  const [error, setError] = useState(false)
  const router = useRouter()

  const loginMutation = useLogin({
    onSuccess: data => {
      if (data.success) {
        setError(false)
        router.push('/create-teams')
        router.refresh() // Force a refresh of the page to update authentication state
      } else {
        setError(true)
      }
    },
    onError: error => {
      setError(true)
      console.error('Login error:', error)
    },
  })

  const handleSubmit = async e => {
    e.preventDefault()
    loginMutation.mutate({ username, password })
  }

  return (
    <form onSubmit={handleSubmit} data-testid="login-form">
      <div className="flex flex-col items-center justify-center mt-20 gap-2">
        <div className="relative z-10 w-[6.25rem] h-[7.8125rem] top-[1.75rem]">
          <Image
            src={LoonsBadge}
            alt="Toronto Walking Soccer Loons Club Logo"
            priority
          />
        </div>
        <div className="flex bg-loonsDarkBrown z-0 w-[17.8125rem] justify-center h-[4.375rem] items-center mb-4">
          <div className="flex items-center justify-center text-2xl border-[0.3125rem] border-loonsRed bg-loonsBrown w-[17.3125rem] text-loonsBeige text-center h-[3.875rem] z-10 font-oswald font-[400] uppercase tracking-wider">
            Loons Team Balancer
          </div>
        </div>
        <div className="flex flex-col justify-center xs:flex-row gap-3 mb-3">
          <input
            type="text"
            value={username}
            onChange={e => setUsername(e.target.value)}
            onFocus={() => setError(false)}
            placeholder="Username"
            required
            disabled={loginMutation.isPending}
            autoComplete="current-username"
            className="border border-gray-300 rounded w-40 h-8 text-center focus:outline-none focus:ring-2 focus:ring-loonsRed disabled:opacity-50"
          />
          <input
            type="password"
            value={password}
            onChange={e => setPassword(e.target.value)}
            onFocus={() => setError(false)}
            placeholder="Password"
            required
            disabled={loginMutation.isPending}
            autoComplete="current-password"
            className="border border-gray-300 rounded w-40 h-8 text-center focus:outline-none focus:ring-2 focus:ring-loonsRed disabled:opacity-50"
          />
        </div>
        <div className="flex justify-center items-center">
          <button
            type="submit"
            disabled={loginMutation.isPending}
            className="border border-gray-400 rounded w-36 h-8 bg-gray-200 hover:bg-gray-300 transition-colors disabled:opacity-50"
          >
            {loginMutation.isPending ? 'Logging in...' : 'Login'}
          </button>
        </div>
      </div>
      <div className="flex justify-center text-center items-center text-loonsRed h-10">
        {error ? "There's been an error. Please try again" : ''}
      </div>
    </form>
  )
}
