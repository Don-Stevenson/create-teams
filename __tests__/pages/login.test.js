import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Login from '../../src/app/login/page'
import LoginForm from '../../src/app/components/features/auth/LoginForm'

jest.mock('../../src/app/components/features/auth/LoginForm', () => {
  return jest.fn(() => (
    <div data-testid="mocked-login-form">Mocked Login Form</div>
  ))
})

describe('Login Page', () => {
  beforeEach(() => {
    LoginForm.mockClear()
  })

  it('renders the login page', () => {
    render(<Login />)
    expect(screen.getByTestId('mocked-login-form')).toBeInTheDocument()
  })

  it('renders the LoginForm component', () => {
    render(<Login />)
    expect(LoginForm).toHaveBeenCalled()
  })

  it('LoginForm is rendered exactly once', () => {
    render(<Login />)
    expect(LoginForm).toHaveBeenCalledTimes(1)
  })

  it('passes correct props to LoginForm', () => {
    render(<Login />)
    expect(LoginForm).toHaveBeenCalledWith({}, {})
  })

  it('handles component re-renders correctly', () => {
    const { rerender } = render(<Login />)
    expect(LoginForm).toHaveBeenCalledTimes(1)

    rerender(<Login />)
    expect(LoginForm).toHaveBeenCalledTimes(2)
  })
})
