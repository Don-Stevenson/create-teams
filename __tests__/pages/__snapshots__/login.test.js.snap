// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page matches snapshot 1`] = `
<div>
  <div
    data-testid="mocked-login-form"
  >
    Mocked Login Form
  </div>
</div>
`;
